index.js

```diff
@@ -263,13 +269,37 @@ module.exports = {
               __frameId: params.frameId,
               _initialPriority: request.initialPriority,
               _priority: request.initialPriority,
-              _initiator: params.initiator.url,
-              _initiator_line: params.initiator.lineNumber,
               pageref: currentPageId,
               request: req,
               time: 0
             };
 
+            switch (params.initiator.type) {
+              case 'script':
+                if (
+                  params.initiator.stack
+                  && params.initiator.stack.callFrames
+                  && params.initiator.stack.callFrames.length > 0
+                ) {
+                  const firstCallFrame = params.initiator.stack.callFrames[params.initiator.stack.callFrames.length - 1];
```

Comment:
Is this really correct? The actual request would be triggered from the top frame (i.e. the most recent) in the call chain, right? To avoid confusion that the first element in the array is the last/latest stack frame, I'd suggest a different name, e.g.  ```const topCallFrame = params.initiator.stack.callFrames[0];```

Reply:
If we want to follow WebPageTest, they put the full initiator object (serialised into a string) into a field called _initiator_detail. Not just for scripts, but for all types. That would give an opportunity to recreate the full stack trace.